ext/AvizExt/viz/taxa_dynamics.jl

```diff
@@ -0,0 +1,273 @@
+
+    g = f[1, 1] = GridLayout()
+
+    _scenarios = copy(scenarios[1:end .∈ [relative_taxa_cover.scenarios], :])
+    scen_groups = if get(opts, :by_RCP, false)
+        ADRIA.analysis.scenario_rcps(_scenarios)
+    else
+        ADRIA.analysis.scenario_types(_scenarios)
+    end
+
+    ADRIA.viz.taxonomy!(
+        g,
+        relative_taxa_cover,
+        scen_groups,
+        opts=opts,
+        axis_opts=axis_opts,
+        series_opts=series_opts
+    )
+
+    return f
+end
+function ADRIA.viz.taxonomy!(
+    g::Union{GridLayout,GridPosition},
+    relative_taxa_cover::YAXArray,
+    scen_groups::Dict{Symbol, BitVector};
+    opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}(),
+    axis_opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}(),
+    series_opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}()
+)::Union{GridLayout,GridPosition}
+    axis_opts[:xlabel] = get(axis_opts, :xlabel, "Year")
+    axis_opts[:ylabel] = get(axis_opts, :ylabel, "Relative Cover")
+
+    show_confints::Bool = get(opts, :show_confints, true)
+    by_functional_groups::Bool = get(opts, :by_functional_groups, true)
+    if by_functional_groups
+        # Create colors
+        n_functional_groups::Int64 = length(relative_taxa_cover.taxa)
+        default_color = Symbol("Set1_" * string(n_functional_groups))
+        color = get(opts, :colors, default_color)
+        _colors = categorical_colors(color, n_functional_groups)
+
+        # Plot results
+        taxonomy_by_intervention!(
+            g,
+            relative_taxa_cover,
+            scen_groups,
+            _colors;
+            show_confints=show_confints,
+            axis_opts,
+            series_opts
+        )
+    else
+        # Use default ADRIA colors for scenario type if use not specified
+        n_scenario_groups::Int64 = length(keys(scen_groups))
+        color = get(opts, :colors, nothing)
+        _colors = isnothing(color) ? [
+            COLORS[scen_name] for scen_name in keys(scen_groups)
+        ] : categorical_colors(color, n_scenario_groups)
+
+        # Plot results
+        intervention_by_taxonomy!(
+            g,
+            relative_taxa_cover,
+            scen_groups,
+            _colors;
+            show_confints=show_confints,
+            axis_opts,
+            series_opts
+        )
+    end
+
+    return g
+end
+
+"""
+    taxonomy_by_intervention!(g::Union{GridLayout, GridPosition}, relative_taxa_cover::YAXArray, scen_groups::Dict{Symbol}, colors::Union{Vector{Symbol},Vector{RGBA{T}}}; show_confints=true, axis_opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}(), series_opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}())::Nothing where {T<:Float32}
+    taxonomy_by_intervention!( ax::Axis, relative_taxa_cover::YAXArray, colors::Union{Vector{Symbol},Vector{RGBA{T}}}; show_confints=true, show_legend=true, series_opts::Dict{Symbol,<:Any}=Dict{Symbol,Any}())::Nothing where {T<:Float32}
+
+Plot relative cover divided between coral taxonomy with each grid cell a different type of
```

Comment:
I think there's a word missing here. "each grid cell ____ a different type of ..."